src/Symfony/Component/Routing/Loader/XmlFileLoader.php

```diff
@@ -118,6 +118,8 @@ protected function parseRoute(RouteCollection $collection, \DOMElement $node, $p
         }
 
         if ($node->hasAttribute('pattern')) {
+            trigger_error('The "pattern" is deprecated since version 2.2 and will be removed in 3.0. Use the "path" option in the route definition instead.', E_USER_DEPRECATED);
```

Comment:
I don't agree as this will let people know both errors at once instead of one after the other.